src/rule-parser.js

```diff
@@ -161,7 +161,11 @@ class EventRequirement extends  Requirement {
     });
   }
 
-  async isFulfilled(context) {
+  async isFulfilled(context, isFirstCycle) {
+    // event requirements should be only checked in first achievement cycle
+    if (!isFirstCycle) {
```

Comment:
Is it possible that an achievement triggers an event, which is skipped here?

Reply:
Good point. Do we want to listen on events we produce? If so, we should probably give the user a possibility to trigger events in the action parts of configuration.